pkg/prometheus/statefulset.go

```diff
@@ -699,9 +705,7 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *Config, ruleConfigMapName
 						SecretKeyRef: p.Spec.Thanos.S3.AccessKey,
 					},
 				})
-			}
-			if p.Spec.Thanos.S3.EncryptSSE != nil && *p.Spec.Thanos.S3.EncryptSSE {
-				thanosArgs = append(thanosArgs, "--s3.encrypt-sse")
+				objstore = append(objstore, "  access_key: $(S3_ACCESS_KEY)")
```

Comment:
I guess those are defined? They must be in pod spec env vars for container.

Reply:
the above of it...

```
                          if p.Spec.Thanos.S3.AccessKey != nil {
				envVars = append(envVars, v1.EnvVar{
					Name: "S3_ACCESS_KEY",
					ValueFrom: &v1.EnvVarSource{
						SecretKeyRef: p.Spec.Thanos.S3.AccessKey,
					},
				})
			}
```

is defiend.